utils/fix-rules.py

```diff
@@ -91,32 +92,31 @@ def find_rules(directory, func):
     results = []
     product_yamls = {}
     product_yaml_paths = {}
+    product_yaml = None
+    product_yaml_path = None
     for root, dirs, files in os.walk(directory):
-        product_yaml = None
-        product_yaml_path = None
 
         if "product.yml" in files:
             product_yaml_path = os.path.join(root, "product.yml")
-            product_yaml = ssg.yaml.open_raw(product_yaml_path)
+            product_yaml = yaml.open_raw(product_yaml_path)
             product_yamls[root] = product_yaml
             product_yaml_paths[root] = product_yaml_path
-            for d in dirs:
-                product_yamls[os.path.join(root, d)] = product_yaml
-                product_yaml_paths[os.path.join(root, d)] = product_yaml_path
+            # for d in dirs:
+            #     product_yamls[os.path.join(root, d)] = product_yaml
+            #     product_yaml_paths[os.path.join(root, d)] = product_yaml_path
         elif root in product_yamls:
             product_yaml = product_yamls[root]
             product_yaml_path = product_yaml_paths[root]
-            for d in dirs:
-                product_yamls[os.path.join(root, d)] = product_yaml
-                product_yaml_paths[os.path.join(root, d)] = product_yaml_path
+            # for d in dirs:
+            #     product_yamls[os.path.join(root, d)] = product_yaml
+            #     product_yaml_paths[os.path.join(root, d)] = product_yaml_path
         else:
             pass
 
         for filename in files:
             path = os.path.join(root, filename)
-            if len(path) < 5 or path[-5:] != '.rule':
+            if len(path) < 8 or path[-8:] != 'rule.yml' or "tests/" in path:
```

Comment:
One thought...

What about providing a content directory walker in the `ssg` module. Something like:

```
for item in ssg.walk_content(types=["product", "rule"]):
    pass
```

Or something similar. Probably out of scope for this PR, but I think it is a common enough occurrence that abstracting away the logic might be nice. Either leave it as a filesystem path or parse the YAML or something... My 2c.

Reply:
Also, please remove magic constants like -8.